TaskModules/powershell/TaskModuleIISManageUtility/AppCmdOnTargetMachines.ps1

```diff
@@ -731,6 +731,35 @@ function Start-Stop-Recycle-ApplicationPool {
     Invoke-VstsTool -Filename $appCmdPath -Arguments $appCmdArgs -RequireExitCodeZero
 }
 
+function ConfigureWebsiteAuthentication {
+    param (
+        [string]$basicAuthentication,
+        [string]$windowsAuthentication,
+        [string]$websiteName
+    )
+
+    Write-Verbose "Configuring website authentication"
+    $appCmdPath, $iisVersion = Get-AppCmdLocation -regKeyPath $AppCmdRegKey
+
+    $anonymousAuthentication = $basicAuthentication -ieq "false" -and $windowsAuthentication -ieq "false"
+
+    $appCmdArgs = [string]::Format('set config "{0}" /section:anonymousAuthentication /enabled:{1} /commit:apphost', $websiteName, $anonymousAuthentication)
+    $command = "`"$appCmdPath`" $appCmdArgs"
+    Write-Verbose "Setting anonymous authentication for website '$websiteName'. Running command $command"
+    Invoke-VstsTool -Filename $appCmdPath -Arguments $appCmdArgs -RequireExitCodeZero
+
+
+    $appCmdArgs = [string]::Format('set config "{0}" /section:basicAuthentication /enabled:{1} /commit:apphost', $websiteName, $basicAuthentication)

```

Comment:
You can have machines where the Basic authentication is not enabled in Windows features. How is the error handled in such cases. Do we guide users appropriately here? This comment can apply to other auth types as well.